### PR TITLE
How to use CORS in Node.js with Express

### LINKED ISSUE #1895: How to use CORS in Node.js with Express
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of the article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your articles a higher ranking.*

How to use Cors in Node.js

### Introduction paragraph (2-3 paragraphs):
CORS stands for Cross-Origin Resource Sharing. Allows us to bypass security applied to a Node.js API. An API is a set procedure for two programs to communicate. This means the two (client and server) have a different origin, i.e., accessing resources from a different server to the server you are in. In this case, trying to make a request to a resource on the server will fail. You are getting something from that server, and it's not the server you (the client) are coming from.

That's a security concern for the browser. These resources are meant to be consumed by other clients and servers. CORS comes into play to disable this mechanism and allow access to these resources. CORS will add a response header [access-control-allow-origins] and specify which origins are to be permitted the access.

CORS helps us to make sure that we are sending the right headers. This is done by sending the headers from the server to the client to inform the browser that the client can access the resources requested.

### Key takeaways: 
What is CORS
How CORS works
Using CORS with Express to access resources from a server

### References: 
https://www.npmjs.com/package/cors
https://expressjs.com/
https://www.npmjs.com/package/cors
https://nodejs.dev/learn/introduction-to-nodejs

### Templates to use as guides
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by kimkimani**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
fixes #1895

### CODE REVIEW

File: articles/how-to-use-cors-in-nodejs-with-express/index.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        "item": "Butter"
+    }
+];
+
+app.get('/ingredients', (req, res) =>{
+    res.send(ingredients);
+});
+app.listen(6069);
+```
+
```
Comment (WanjaMIKE) [inline articles/how-to-use-cors-in-nodejs-with-express/index.md:80]: Could you explain more on the above code snippets

File: articles/how-to-use-cors-in-nodejs-with-express/index.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
+Node.js is an open-source and cross-platform runtime for executing JavaScript code on the server-side. One of the popular Node.js server frameworks is Express. Express allows you to configure and manage an HTTP server to access resources from the same origin they as being served (i.e., the same domain).
+
+These three parts form an origin.
```
Comment (WanjaMIKE) [inline articles/how-to-use-cors-in-nodejs-with-express/index.md:3]: Add a link to help the reader learn more on `origin`

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
@kimkimani Nice article. Just minor changes

**Comment by WanjaMIKE** [comment]
Nice article @kimkimani I have learned a lot about using CORS in Node.js and Express

**Review by WanjaMIKE — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
great work gents @kimkimani 

cc @WanjaMIKE

**Pull request merged**